src/o365/graph/commands/teams/teams-channel-set.ts

```diff
@@ -0,0 +1,163 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../GraphCommand';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelName: string;
+  newChannelName?: string;
+  description?: string
+}
+class GraphTeamsChannelSetCommand extends GraphCommand{
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_SET}`;
+  }
+  public get description(): string {
+    return 'Updates properties of a specified channel in the given Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.newChannelName = typeof args.options.newChannelName !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    return telemetryProps;
+  }
+
+   public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+    .ensureAccessToken(auth.service.resource, cmd, this.debug)
+    .then((): request.RequestPromise => {
+    const getrequestOptions: any = {
+      url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels?$filter=displayName eq '${encodeURIComponent(args.options.channelName as string)}'`,
+      headers: Utils.getRequestHeaders({
+        authorization: `Bearer ${auth.service.accessToken}`,
+        accept: 'application/json;odata.metadata=none'
+      }),
+      json: true
+    }
+    if (this.debug) {
+      cmd.log('Executing web request...');
+      cmd.log(getrequestOptions);
+      cmd.log('');
+    }
+    return request.get(getrequestOptions);
+  })
+  .then((res:{ value: Channel[] }): request.RequestPromise => {
+      let channelId: string = '';
+      if (this.debug) {
+        cmd.log('Response:');
+        cmd.log(res);
+        cmd.log('');
+      }
+       channelId = res.value[0].id;
```

Comment:
It is possible that no channel with the specified name exists. We should catch this rather than throwing a null reference exception.